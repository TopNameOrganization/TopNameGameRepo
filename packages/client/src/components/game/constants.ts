export const TileSize = 32;
export enum Tile {
  Empty = 0,
  Brick = 1,
  Concrete = 2,
  Stair = 3,
  Rope = 4,
  Bonus = 5,
  Enemy = 6,
  Player = 7,
  Out = 100,
}

export const VELOCITY = 200;
export enum Orientation {
  Left = 0,
  Right = 1,
}
export enum RunnerAction {
  Stay = 0,
  Fall = 1000,
  MoveLeft = 37,
  MoveUp = 38,
  MoveRight = 39,
  MoveDown = 40,
}

export enum AnimationPhase {
  Stay = 0,
  Run = 1,
  Climb = 2,
  Hang = 3,
  Fall = 4,
}
