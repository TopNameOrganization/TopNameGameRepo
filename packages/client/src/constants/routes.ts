export const ROUTES = {
  root: '/',
  home: '/home',
  game: '/game',
  login: '/login',
  signup: '/signup',
  profile: '/profile',
  leaderBoard: '/leader-board',
  forum: '/forum',
  forumCreate: '/forum/create',
  forumDetail: '/forum/:id',
}
