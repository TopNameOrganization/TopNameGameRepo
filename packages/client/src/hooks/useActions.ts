import {useDispatch} from "react-redux";
import {bindActionCreators} from "@reduxjs/toolkit";
import Actions from '../store/actions/index'

export const useActions = () => {
  const dispatch = useDispatch()
  return bindActionCreators(Actions, dispatch)
}
