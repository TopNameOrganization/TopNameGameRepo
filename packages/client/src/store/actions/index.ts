import * as UserAction from './userAction'
import * as UserProfileAction from './changeUserProfileAction'

export default {
  ...UserAction,
  ...UserProfileAction
}
