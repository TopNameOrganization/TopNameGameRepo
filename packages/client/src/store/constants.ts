import {UserState} from "./types";

export const initialState: UserState = {
  user: {
    id: null,
    first_name: '',
    second_name: '',
    display_name: '',
    login: '',
    email: '',
    phone: '',
    avatar: ''
  },
  loading: false,
  error: null
}
