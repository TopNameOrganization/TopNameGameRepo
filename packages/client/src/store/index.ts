import { rootReducer, RootState } from './reducers'
import { configureStore } from "@reduxjs/toolkit";

export const setupStore = (initialState?: RootState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState: initialState
  });
}
