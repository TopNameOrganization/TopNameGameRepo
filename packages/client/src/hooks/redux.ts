import {TypedUseSelectorHook, useDispatch, useSelector} from "react-redux";
import {AppDispatch, RootState} from "../store/reducers";
import {bindActionCreators} from "@reduxjs/toolkit";
import Actions from "../store/actions";

export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export const useAppActions = () => bindActionCreators(Actions, useAppDispatch())
