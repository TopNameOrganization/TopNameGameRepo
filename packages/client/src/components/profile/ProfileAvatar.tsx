import * as React from 'react';
import Box from '@mui/material/Box';
import Avatar from "@mui/material/Avatar";
import ProfileAvatarDialog from './ProfileAvatarDialog';
import {useAppSelector} from "../../hooks/redux";

const resourcesUrl = 'https://ya-praktikum.tech/api/v2/resources'

export default function ProfileAvatar() {
    const [isShow, setIsShow] = React.useState<boolean>(false);
    const { avatar, display_name } = useAppSelector((state) => state.userReducer.user);


  const handleMouseEnter = () => {
    setIsShow(true)
  }
  const handleMouseLeave = () => {
    setIsShow(false)
  }

  return (
    <Box
      sx={{ position: 'relative', borderRadius: '50%', overflow: 'hidden' }}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}>
      <Avatar
        alt={display_name || 'Nick name'}
        src={resourcesUrl + avatar || ''}
        sx={{ width: 100, height: 100, m: 1 }}
      />
      <ProfileAvatarDialog isShowDialogButton={isShow} />
    </Box>
  )
}
