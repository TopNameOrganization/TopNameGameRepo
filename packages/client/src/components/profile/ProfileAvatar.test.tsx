import * as renderer from 'react-test-renderer';
import ProfileAvatar from './ProfileAvatar';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import * as appHooks from "../../hooks/redux";
import { Provider } from "react-redux";
import { initialState } from "../../store/constants";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const useAppSelectorMock = jest.spyOn(appHooks, 'useAppSelector');

describe('Profile', () => {
  it('should renders profile correctly', () => {
    useAppSelectorMock.mockReturnValue(initialState)
    const store = mockStore(initialState)
    const tree = renderer
        .create(
            <Provider store={store}>
                <ProfileAvatar />
            </Provider>)
        .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
