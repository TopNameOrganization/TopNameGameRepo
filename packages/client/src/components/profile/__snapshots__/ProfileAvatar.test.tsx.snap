// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should renders profile correctly 1`] = `
<div
  className="MuiBox-root css-1142v5w"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiAvatar-root MuiAvatar-circular css-ghc07c-MuiAvatar-root"
  >
    <img
      alt="Nick name"
      className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
      src="https://ya-praktikum.tech/api/v2/resourcesundefined"
    />
  </div>
</div>
`;
